src/main.cpp

```diff
@@ -543,3 +550,25 @@ void exit_handler(int s)
         exit( exit_status );
     }
 }
+
+void hangup_handler(int signum)
```

Comment:
The parameter names were an oversight on my part.  Will be fixed.

As for creating a separate function and calling it from both handlers, this should be possible and probably would be better.  I'll give it a try and see how it looks